cdap-app-fabric/src/main/java/io/cdap/cdap/app/deploy/AppDeletionPublisher.java

```diff
@@ -85,5 +82,16 @@ public void publishAppDeletionEvent(ApplicationId appId) {
         return null;
         },
       retryStrategy);
+      Retries.supplyWithRetries(() -> {
+        try {
+          messagingContext.getMessagePublisher().publish(NamespaceId.SYSTEM.getNamespace(), topic.getTopic(),
+                                                         GSON.toJson(message));
+        } catch (TopicNotFoundException | ServiceUnavailableException e) {
+          throw new RetryableException(e);
+        } catch (IOException | AccessException e) {
+          throw Throwables.propagate(e);
+        }
+        return null;
+        }, retryStrategy, t -> t instanceof RetryableException);
   }
```

Comment:
@tivv updated `supplyWithRetries` logic

Reply:
* I see two publish calls, did you mean it?
 * With the predicate you should not need try/catch, it should be handled by the predicate. Please remove try/catch

Reply:
oh thats a mistake, let me correct it.
But L92 is not retryable ?

Reply:
Yes, remove the whole try/catch unless there are other reasons to have it. But generally you should not convert AccessException to something else because this would change your HTTP codes from 403 to 500

Reply:
so this is not allowed (still see unhandled exceptions for publish: 
``` 
Retries.supplyWithRetries(() -> {
          messagingContext.getMessagePublisher().publish(NamespaceId.SYSTEM.getNamespace(), topic.getTopic(),
                                                         GSON.toJson(message));
          return null;
        }, retryStrategy, t -> t instanceof TopicNotFoundException || t instanceof IOException || 
        t instanceof AccessException || t instanceof ServiceUnavailableException); 

```

Reply:
Well, you should propagate AccessException all the way up